module/VuFind/src/VuFind/Controller/UpgradeController.php

```diff
@@ -779,11 +779,14 @@ public function fixmetadataAction()
 
         // Process submit button:
         if ($this->formWasSubmitted()) {
-            $converter = $this->serviceLocator->get(Converter::class);
+            $resourceService = $this->getDbService(ResourceServiceInterface::class);
+            $resourcePopulator = $this->serviceLocator->get(ResourcePopulator::class);
             foreach ($problems as $problem) {
                 try {
                     $driver = $this->getRecordLoader()->load($problem->getRecordId(), $problem->getSource());
-                    $problem->assignMetadata($driver, $converter)->save();
+                    $resourceService->persistEntity(
+                        $resourcePopulator->assignMetadata($problem, $driver)
```

Comment:
We could potentially add an `assignAndPersistMetadata` wrapper method to simplify this, if you think it's worth the effort.

Reply:
Looks like there is only this one case where both are needed, so maybe not worth the effort.